string name
string node
---
bool done